.github/workflows/upload-metadata-files.yml

```diff
@@ -0,0 +1,52 @@
+name: Upload any Changed Metadata Files
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - "airbyte-integrations/connectors/**/metadata.yaml"
+
+jobs:
+  deploy-catalog-to-stage:
+    name: "Upload Metadata Files to Metadata Service"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Airbyte Cloud
+        uses: actions/checkout@v2
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v35
+        with:
+          files: "airbyte-integrations/connectors/**/metadata.yaml"
+      - name: Setup Python 3.10
+        if: steps.changed-files-specific.outputs.any_changed == 'true'
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Install metadata_service
+        if: steps.changed-files-specific.outputs.any_changed == 'true'
```

Comment:
nit: you could add a previous step to cancel the workflow run `steps.changed-files-specific.outputs.any_changed == 'false'`